src/community.c

```diff
@@ -3567,3 +3567,1100 @@ int igraph_i_compare_communities_rand(const igraph_vector_t *v1,
 
   return IGRAPH_SUCCESS;
 }
+
+/* Clean membership.
+ *
+ * This function ensures that the membership vector is renumbered in consecutive
+ * numbers 0, ..., q, where q is the number of distinct clusters. The number of
+ * distinct clusters is also returned in \c nb_clusters.
+ */
+int igraph_i_community_leiden_clean_membership(igraph_vector_t *membership, igraph_integer_t* nb_clusters)
```

Comment:
I think we already have this functionality in `igraph_reindex_membership()`.

Reply:
Yes, true.

I though the existing method was a bit inefficient because of the unnecessary sorting that is done.

But if you think it is better to use the already existing functions I can do so. From a maintenance point of view you would be right I think.

I could also make the method more efficient (but use the same call signature of course). But my current implementation makes some assumptions, which may not be warranted in all function uses. That is why I didn't touch it originally. I could take a closer look.

Reply:
I'd rather have a bit more inefficient implementation than the duplication of functionality across the codebase. It could also be an option to make `igraph_reindex_membership()` more efficient by replacing its implementation with yours.